hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleCreateTransitionLogic.java

```diff
@@ -50,12 +61,60 @@ public void doStateTransition() {
         }
     }
 
-    private void transitionFor(ScheduleCreateTransactionBody op) {
-        // TODO: Implement transitionFor() functionality
+    private void transitionFor(ScheduleCreateTransactionBody op) throws DecoderException {
+        var scheduledTXPayer = op.hasPayer() ? op.getPayer() : txnCtx.activePayer();
+        var schedule = store.getScheduleID(op.getTransactionBody().toByteArray(), scheduledTXPayer);
+        if (schedule.isEmpty()) {
+
+            var bytes = op.getTransactionBody().toByteArray();
+            var payer = Optional.of(op.getPayer());
+            var schedulingAccount = txnCtx.activePayer();
+            var now = RichInstant.fromJava(txnCtx.consensusTime());
+            Optional<JKey> adminKey = Optional.empty();
+            if (op.hasAdminKey()) {
+                adminKey = asUsableFcKey(op.getAdminKey());
+            }
+
+            var result = store.createProvisionally(
```

Comment:
Since on line `65` we have the `scheduledTXPayer` we can change the `createProvisionally` method from `Optional<AccountID> payer` to `AccountID payer`. We won't need to have `line 70` and the checks in `HederaScheduleStore` for the optional payer `payer.isPresent()`